AwooChat.py

```diff
@@ -29,84 +29,76 @@ def titleWatchdog(msg):
-            if i in alertList:
-                for j in parsedPokemon[i]:
-                    message += AwooUtils.getPokemonName(i) + " has been spotted! It despawns at " + str(j['despawn']) + \
-                            ". [ᕕ( ᐛ )ᕗ](http://maps.google.com?q=" + str(j['lat']) + "," + str(j['lon']) + ")\n"
-
-        if (chat_id < 0):
-            # Group chat, so we need to handle this
-            now_time = now.time()
-            if now_time >= time(06, 00) and now_time <= time(23, 59):
-                if len(message) != 0:
-                    if (len(message) > 4095):
-                        messages = message.split('\n')
+        if source not in pokemonBySource
+            pokemonBySource[source] = parsePokemon(source)
+        parsedPokemon = pokemonBySource[source]
+        message = ""
+
+        for i in set(alertList) & set(parsedPokemon.keys()):
+            for j in parsedPokemon[i]:
+                message += AwooUtils.getPokemonName(i) + \
+                        " has been spotted! It despawns at " + \
+                        str(j['despawn']) + \
+                        ". [ᕕ( ᐛ )ᕗ](http://maps.google.com?q=" + \
+                        str(j['lat']) + "," + str(j['lon']) + ")\n"
+
+        if len(message) != 0 and \
+                # Send message to private chat any time
+                (chat_id >= 0 or \
+                # Don't spam group chat before 6am
+                    (time(06, 00) <= now_time <= time(23, 59))):
+            if len(message) > 4095:
+                messages = message.split('\n')
+                currMsg = ""
+                for m in messages:
+                    if len(currMsg) + len(m) + 1 > 4095:
+                        bot.sendMessage(chat_id, currMsg, "Markdown", True)
                         currMsg = ""
-                        for m in messages:
-                            if len(currMsg) + len(m) + 1 > 4095:
-                                bot.sendMessage(chat_id, currMsg, "Markdown", True)
-                                currMsg = ""
-                            else:
-                                currMsg += m + "\n"
                     else:
-                        bot.sendMessage(chat_id, message, "Markdown", True)
-        else:
-            if len(message) != 0:
-                if (len(message) > 4095):
-                    messages = message.split('\n')
-                    currMsg = ""
-                    for m in messages:
-                        if len(currMsg) + len(m) + 1 > 4095:
-                            bot.sendMessage(chat_id, currMsg, "Markdown", True)
-                            currMsg = ""
-                        else:
-                            currMsg += m + "\n"
-                else:
-                    bot.sendMessage(chat_id, message, "Markdown", True)
+                        currMsg += m + "\n"
+                if currMsg:
+                    bot.sendMessage(chat_id, currMsg, "Markdown", True)
```

Comment:
I added this to make sure that messages are not lost. Previously, if a message was longer than 4095 chars then any extra after the 4095 would be lost because it would only print when it reset.